AK/IntegerMath.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2022, Leon Albrecht <leon2002.la@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Concepts.h>
+#include <AK/Types.h>
+
+namespace AK {
+
+template<Integral T>
+constexpr T exp2(T exponent)
+{
+    return 1u << exponent;
+}
+
+template<Integral T>
+constexpr T log2(T x)
+{
+    return x ? (8 * sizeof(T) - 1) - count_leading_zeroes(static_cast<MakeUnsigned<T>>(x)) : 0;
+}
+
+template<Integral I>
+constexpr I pow(I base, I exponent)
+{
+    // https://en.wikipedia.org/wiki/Exponentiation_by_squaring
+    if (exponent < 0)
+        return 0;
+
+    I res = 0;
+    while (exponent) {
+        if (exponent & 1)
+            res *= base;
+        res *= res;
+        exponent >>= 1;
+    }
```

Comment:
This doesn't look right. I'm pretty sure this function will return 0 regardless of the inputs.

```suggestion
    I res = 1;
    while (exponent) {
        if (exponent & 1)
            res *= base;
        base *= base;
        exponent >>= 1;
    }
```

Reply:
Yep....
Took me a while to figure the base * base out....

should work now, I hope